src/collision/cm_test.h

```diff
@@ -34,13 +32,51 @@ static inline float Cm_DistanceToPlane(const vec3_t point, const cm_bsp_plane_t
 
 int32_t Cm_PlaneTypeForNormal(const vec3_t normal);
 int32_t Cm_SignBitsForNormal(const vec3_t normal);
-cm_bsp_plane_t Cm_TransformPlane(const mat4_t *matrix, const cm_bsp_plane_t *plane);
+
+/**
+ * @brief
+ */
+static inline cm_bsp_plane_t Cm_Plane(const vec3_t normal, float dist) {
+
+	return (cm_bsp_plane_t) {
+		.normal = normal,
+		.dist = dist,
+		.type = Cm_PlaneTypeForNormal(normal),
+		.sign_bits = Cm_SignBitsForNormal(normal)
+	};
+}
+
+/**
+ * @brief
+ */
+static inline cm_bsp_plane_t Cm_TransformPlane(const mat4_t matrix, const cm_bsp_plane_t *plane) {
+	const vec4_t out = Mat4_TransformQuakePlane(matrix, plane->normal, plane->dist);
+	return Cm_Plane(Vec4_XYZ(out), out.w);
+}
+
 int32_t Cm_BoxOnPlaneSide(const vec3_t mins, const vec3_t maxs, const cm_bsp_plane_t *plane);
 _Bool Cm_PointInsideBrush(const vec3_t point, const cm_bsp_brush_t *brush);
 int32_t Cm_SetBoxHull(const vec3_t mins, const vec3_t maxs, const int32_t contents);
 int32_t Cm_PointLeafnum(const vec3_t p, int32_t head_node);
 int32_t Cm_PointContents(const vec3_t p, int32_t head_node);
-int32_t Cm_TransformedPointContents(const vec3_t p, int32_t head_node, const mat4_t *inverse_matrix);
+
+/**
+ * @brief Contents check for non-world models. Rotates and translates the point
+ * into the model's space, and recurses the BSP tree. For inline BSP models,
+ * the head node is the root of the model's subtree. For mesh models, a special
+ * reserver box hull is used.
+ *
+ * @param p The point, in world space.
+ * @param head_hode The BSP head node to recurse down.
+ * @param inverse_matrix The inverse matrix of the entity to be tested.
+ *
+ * @return The contents mask at the specified point.
+ */
+static inline int32_t Cm_TransformedPointContents(const vec3_t p, int32_t head_node, const mat4_t inverse_matrix) {
+	const vec3_t p0 = Mat4_Transform(inverse_matrix, p);
+	return Cm_PointContents(p0, head_node);
+}
+
```

Comment:
I'm not sure I like this in the header, either.

Reply:
I'm just worried these won't inline properly without it, and passing a mat4 by value might be expensive, but it's less ugly/consistent with our vector system where we don't pass them via pointer. I can move them back though, and we can see what happens.